cmd/juju/application/utils/names.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package utils
+
+import (
+	"regexp"
+	"strings"
+	"unicode"
+
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+)
+
+var alphaChar = regexp.MustCompile("[A-Z]")
+var tailNumberSuffix = regexp.MustCompile("-[0-9]+$")
+
+// ValidateApplicationName takes a name and attempts to validate the application
+// name, before returning a reason why it's not valid.
+func ValidateApplicationName(name string) error {
+	if !names.IsValidApplication(name) {
```

Comment:
You could invert the check to return early if it is valid and remove one level of indentation from the error checking code.